deployments/Dockerfiles/localnode/app.py

```diff
@@ -218,7 +218,10 @@ def hard_reset() -> Tuple[Any, int]:
             if IS_DEV_MODE:
                 period_dumper.dump_period()
 
-            tendermint_node.prune_blocks()
+            return_code = tendermint_node.prune_blocks()
+            if return_code:
+                tendermint_node.start(start_monitoring=perform_monitoring)
+                raise RuntimeError("Could not perform `unsafe-reset-all` successfully!")
```

Comment:
We now raise if `prune_block` fails and restart the node. The error will be handled below and the status will be `False`, accompanied by a failure message and the exception's message.